jcli/src/jcli_app/vote/decryption_share.rs

```diff
@@ -0,0 +1,69 @@
+use super::Error;
+use crate::jcli_app::utils::OutputFormat;
+use chain_vote::{OpeningVoteKey, Tally};
+use serde::Serialize;
+use std::{
+    fs::File,
+    io::{stdin, Read},
+    path::PathBuf,
+};
+use structopt::StructOpt;
+
+/// Create the decryption share for decrypting the tally of private voting.
+/// The outputs are provided as hex-encoded byte sequences.
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct TallyDecryptionShare {
+    /// The path to hex-encoded encrypted tally state. If this parameter is not
+    /// soecified, the encrypted tally state will be read from the standard
+    /// input.
+    encrypted_tally: Option<PathBuf>,
+    /// The path to hex-encoded decryption key.
+    decryption_key: PathBuf,
+    #[structopt(flatten)]
+    output_format: OutputFormat,
+}
+
+#[derive(Serialize)]
+struct Output {
+    state: String,
+    share: String,
+}
+
+impl TallyDecryptionShare {
+    pub fn exec(&self) -> Result<(), Error> {
+        let encrypted_tally_hex = if let Some(encrypted_tally_path) = &self.encrypted_tally {
+            let mut data = Vec::new();
+            File::open(encrypted_tally_path)?.read_to_end(&mut data)?;
+            data
+        } else {
+            let mut data = Vec::new();
+            stdin().read_to_end(&mut data)?;
+            data
+        };
```

Comment:
did you consider `jcli_app::io::read_line`?

Reply:
I didn't but it looks like what I need here, thanks.